### PR TITLE
Add ToS marketing opt out page (Fixes #15946)

### LINKED ISSUE #15946: ToS opt-out for Firefox Desktop users that come from paid acquisition.
### Description 
Provide opt-outs for marketing data on the Download page to users coming from paid campaigns to improve our transparency and align with our updated data policy & principles. 

### Technical details 
See this message https://mozilla.slack.com/archives/C07URCVARC7/p1738172272626799?thread_ts=1738140474.995049&cid=C07URCVARC7

### Brief link
https://docs.google.com/document/d/1K3nvg4vooMEnCyjU__Nq5QI_NPu5gyhVTCejH-Ub5M0/edit?tab=t.0#heading=h.lsmbopvw8vhv

### Design assets
https://www.figma.com/design/B0qd9saaS0LNnuMOlIALSE/TOS-Firefox-Desktop?node-id=53-211&t=k1JjyiLSacAGNn9U-1

### Languages 
we should translate to all the same languages that /new is localized in

### URL 
/firefox/landing/get

### PR DESCRIPTION
**Pull request by alexgibson**
## One-line summary

- Adds a custom Firefox landing page that will be used as a destination for people who click on ad links.
- Also updates a couple of strings on `/firefox/new/`, at request of the product team.

## Significant changes and points to review

The "opt-out" checkbox is effectively a full opt-out of analytics and attribution in general, using the same cookie as our consent banner / cookie settings page. The only new detail is that because it is "opt-out" and not "opt-in", when someone unchecks the input, we must "undo" things like removing attribution parameters from download links and removing the existing attribution cookie.

The checkbox will only appear if:

- Visitors are outside of EU/EAA (e.g. they are in the US).
- The do not have DNT / GPC enabled.
- They have not already explicitly rejected analytics cookies (e.g. via the `/privacy/websites/cookie-settings` page, or unchecking the checkbox previously).

## Issue / Bugzilla link

#15946

## Testing

http://localhost:8000/en-US/firefox/landing/get/

### CODE REVIEW

File: tests/unit/spec/base/stub-attribution/stub-attribution.js [hunk 1]
```diff
@@ -1205,37 +1205,174 @@ describe('stub-attribution.js', function () {
...
+            expect(Mozilla.StubAttribution.getGtagClientID()).toBeNull();
+        });
     });
 
-    it('should return a null if accessing GTAG API throws an error', function () {
-        window.dataLayer = sinon.stub().throws(function () {
-            return new Error();
+    describe('removeAttributionData', function () {
+        beforeEach(function () {
+            const winUrl =
```
Comment (maureenlholland) [inline tests/unit/spec/base/stub-attribution/stub-attribution.js:1246]: **curious (non-blocking)**: how do you know all the links to check here?

**Review by maureenlholland — COMMENTED** [review]
very minor question and nits on in-progress review

**Review by maureenlholland — CHANGES_REQUESTED** [review]
The built-in tests are excellent and the logic breaks down really cleanly 🥇 

Successfully tested with:

- geo US (show checkbox)
- geo FR (no checkbox or attribution)
- GPC setting on (no checkbox or attribution)
- Previously rejected (no checkbox or attribution)
- Chrome 133, Safari 18.1.1, Edge 132, Firefox 135
   - checking one checkbox checks them all
   - unchecking one checkbox unchecks them all
   - unchecking sets consent pref cookie
   - checking sets attribution cookies

The only blocking issue I have here is for IE 10/11 and it might just be misunderstanding of expectation. I see no checkbox to opt out of attribution (see comment). If it seems like this is a bug, we might need to check on an actual windows computer (I tested on browserstack, and none of my console logs showed up, don't know if the file wasn't running because of IE or browserstack)

File: bedrock/firefox/templates/firefox/landing/get.html [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ This Source Code Form is subject to the terms of the Mozilla Public
+ License, v. 2.0. If a copy of the MPL was not distributed with this
+ file, You can obtain one at https://mozilla.org/MPL/2.0/.
+#}
+
+{% extends "firefox/new/desktop/download.html" %}
+
+{% block canonical_urls %}<meta name="robots" content="noindex,follow">{% endblock %}
+
+{% block experiments %}{% endblock %}
```
[the remark this answers is not in the document]
Comment (alexgibson) [reply bedrock/firefox/templates/firefox/landing/get.html:11]: This will basically ensure that if any experiment are run in `firefox/new/desktop/download.html`, overriding the block here will ensure the experiment won't also trigger on this custom page.

File: bedrock/firefox/templates/firefox/landing/get.html [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+{% block extrahead %}
+  {{ super() }}
+  {{ css_bundle('firefox_marketing_opt_out') }}
+{% endblock %}
+
+{% macro marketing_opt_out_checkbox(id='marketing-opt-out') -%}
+  <label for="{{ id }}" class="marketing-opt-out-checkbox-label hidden">
+    <input type="checkbox" id="{{ id }}" class="marketing-opt-out-checkbox-input">
+    <span class="marketing-opt-out-text">
+      {% set attrs = 'href="https://support.mozilla.org/kb/marketing-data" target="_blank" rel="noopener"' %}
```
[the remark this answers is not in the document]
Comment (alexgibson) [reply bedrock/firefox/templates/firefox/landing/get.html:22]: Yeah good call out, but I don't think this is a blocker. No traffic will go to this page until everything is ready.

File: tests/unit/spec/firefox/landing/marketing-opt-out.js [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+ * Jasmine docs: https://jasmine.github.io/
+ * Sinon docs: http://sinonjs.org/docs/
+ */
+
+import MarketingOptOut from '../../../../../media/js/firefox/landing/get/marketing-opt-out.es6';
+
+describe('marketing-opt-out.es6.js', function () {
+    beforeEach(function () {
+        const optOut = `<div id="opt-out">
+            <label for="marketing-opt-out-primary" class="marketing-opt-out-checkbox-label">
```
[the remark this answers is not in the document]
Comment (alexgibson) [reply tests/unit/spec/firefox/landing/marketing-opt-out.js:17]: Good suggestion 👍

File: tests/unit/spec/base/stub-attribution/stub-attribution.js (hunk 1 above)
Comment (alexgibson) [reply tests/unit/spec/base/stub-attribution/stub-attribution.js:1246]: I just copied it from the stub attribution tests 😉

**Review by maureenlholland — APPROVED** [review]
r+ ⚙️
